LanguageFeatures/Patterns/shared_case_scope_A01_t02.dart

```diff
@@ -62,6 +62,8 @@ main() {
       print(a);
       break;
     case final int a when a == 42:
+//                 ^
+// [cfe] unspecified
```

Comment:
I thought we should have this error message, but at this point I do not think we should have it: There is no error just because we have different declarations of `a` with different types/finalities, it's only an error to _use_ that variable in the body.

So there's no error here, either.